### ISSUE TITLE
Crash when extract frames

### ISSUE THREAD

**Issue by khs123** [dmlc/decord#113]
I used decord installed with my GPU to get frames from the video, everything is fine, but now I have a problem with a strange clip, the size of the clip is only about 1.5mb. When I use decord to extract clip, the decord raise error is as follows
`terminate called after throwing an instance of 'std::length_error'
what():  vector::reserve
`
My code
`
video_path = os.path.normpath(video_path)  #
        rotateCode = check_rotation(video_path)
        ctx = cpu(0)
        if os.environ.get('PRODUCTION'):
            ctx = gpu(0)
        vr = VideoReader(video_path, ctx=ctx)

        if start < 0:
            start = 0
        if end < 0:
            end = len(vr)

        frames_list = list(range(start, end, every))
        if every > 25 and len(frames_list) < 1000:
            frames = vr.get_batch(frames_list).asnumpy()
            for index, frame in zip(frames_list, frames):
                if os.environ.get('PRODUCTION'):
                    frame = correct_rotation(frame, rotateCode)
        
`
My video
`
https://drive.google.com/file/d/1JJ38REU5dW7AtmVS3y9nXg-gWct_4Bxd/view?usp=sharing
`

**Comment by zhreshold** [comment]
There's a bug when video(e.g., the one you attached) has invalid duration/frame-count, the overflow occurs.
For now I can only fix the bound issue, but for those raw webm videos with insane metadata, video reader can only throw an error instead of fixing it which may be an overkill.

```
Input #0, matroska,webm, from 'liveness.mp4':
  Metadata:
    encoder         : Chrome
  Duration: N/A, start: 0.000000, bitrate: N/A
    Stream #0:0(eng): Video: vp9 (Profile 0), yuv420p(tv), 630x480, SAR 1:1 DAR 21:16, 1k tbr, 1k tbn, 1k tbc (default)
    Metadata:
      alpha_mode      : 1
```

Fixed is included in #119

**Comment by khs123** [comment]
Thank you bro ! I will try #119
